pkg/tools/fs/utils_test.go

```diff
@@ -26,55 +27,88 @@ import (
 	"testing"
 	"time"
 
+	"github.com/google/uuid"
 	"github.com/stretchr/testify/require"
 
 	"github.com/networkservicemesh/sdk/pkg/tools/fs"
 )
 
-func Test_WatchFile(t *testing.T) {
-	t.Skip("https://github.com/networkservicemesh/sdk/issues/848")
+func checkPathError(t *testing.T, err error) {
+	if err == nil {
+		return
+	}
 
+	if fErr, ok := err.(*iofs.PathError); ok {
+		if u := fErr.Unwrap(); u != nil {
+			t.Log(u) // Without unwrapping printing gives only numeric error code.
+		}
+	}
+	require.Nil(t, err)
+}
+
+func Test_WatchFile(t *testing.T) {
 	root := filepath.Join(os.TempDir(), t.Name())
+
+	path := filepath.Join(root, uuid.New().String())
+	err := os.MkdirAll(path, os.ModePerm)
 	defer func() {
-		_ = os.RemoveAll(root)
+		_ = os.RemoveAll(path)
 	}()
-
-	path := filepath.Join(root, "A")
-	_ = os.MkdirAll(path, os.ModePerm)
+	checkPathError(t, err)
 
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
 	filePath := filepath.Join(path, "file1.txt")
 	ch := fs.WatchFile(ctx, filePath)
 
-	expectEvent := func(assertFunc func(require.TestingT, interface{}, ...interface{})) {
+	expectEvent := func() []byte {
 		select {
 		case <-time.After(time.Second):
 			debug.PrintStack()
-			t.Fatal("no events")
-		case update := <-ch:
-			assertFunc(t, update)
+			require.Fail(t, "timeout waiting for event", filePath)
+		case event := <-ch:
+			return event
 		}
+		return nil
 	}
 
-	expectEvent(require.Nil)
+	require.Nil(t, expectEvent(), filePath) // initial file read, nil because file doesn't exist
 
-	require.Nil(t, ioutil.WriteFile(filePath, []byte("data"), os.ModePerm))
-
-	expectEvent(require.NotNil)
+	err = ioutil.WriteFile(filePath, []byte("data"), os.ModePerm)
+	checkPathError(t, err)
+	require.NotNil(t, expectEvent(), filePath) // file created
 
+	macOSName := "darwin"
 	// https://github.com/fsnotify/fsnotify/issues/11
-	if runtime.GOOS != "darwin" {
-		expectEvent(require.NotNil)
+	if runtime.GOOS != macOSName {
+		require.NotNil(t, expectEvent(), filePath) // file write. Go on MacOS doesn't support write events
 	}
 
-	_ = os.RemoveAll(root)
-
-	expectEvent(require.Nil)
+	err = os.RemoveAll(path)
+	checkPathError(t, err)
+	if runtime.GOOS != macOSName {
+		require.Nil(t, expectEvent(), filePath) // file removed
+	} else {
+		event := expectEvent()
+		if event != nil { // ...but sometimes we still get write events on MacOS. Very rarely, about 1/500 test retries
+			event = expectEvent()
+		}
+		require.Nil(t, event, filePath)
+	}
 
-	_ = os.MkdirAll(path, os.ModePerm)
-	require.Nil(t, ioutil.WriteFile(filePath, []byte("data"), os.ModePerm))
+	err = os.MkdirAll(path, os.ModePerm)
```

Comment:
If you modify `checkPathError` to simply log and return an error instead of making `require` stuff, you can replace this loop with:
```go
require.Eventually(t, func() {
    return checkPathError(os.MkdirAll(path, os.ModePerm)) == nil
}, timeout, tick)
```

Reply:
`require.Eventually` waits `tick` time before first condition check, which makes the test unnecessarily slower on average, so we can't reduce this code to only `require.Eventually`.
But it will probably be more readable than this raw loop, so I replaced it.